java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -199,6 +200,9 @@ private void setValueFromParser(ValueVector valueVector, int i) throws IOExcepti
     case FLOAT8:
       ((Float8Vector)valueVector).getMutator().set(i, parser.readValueAs(Double.class));
       break;
+    case VARBINARY:
+      ((VarBinaryVector)valueVector).getMutator().setSafe(i, parser.readValueAs(String.class).getBytes(UTF_8));
+      break;
```

Comment:
Right -- the integration tests are only generating ASCII strings so this works for now. Hexadecimal would be better. I can do this in Python / C++ if you have some Java code to convert to `byte[]` from a hex string